spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/AcknowledgeablePubsubMessage.java

```diff
@@ -38,9 +39,26 @@
 	 */
 	String getSubscriptionName();
 
+	/**
+	 * Acknowledge the message asynchronously
+	 */
+	void ackAsync();
```

Comment:
That's all OK, but I see that downstream API is like this:
```
 public ApiFuture<ResponseT> futureCall(RequestT request) {
```

Shouldn't we expose a `org.springframework.util.concurrent.ListenableFuture` as a return type for possible use-cases when end-user would like to react for the call result or failure?

Reply:
How would we turn `ApiFuture` into `org.springframework.util.concurrent.ListenableFuture`? It doesn't look like `ApiFuture` has an error callback support.

Reply:
Never mind, I see we did it on the publish side.